### PR TITLE
Feature/refactoring

### PR DESCRIPTION
**Pull request by mzanrosso**
[StyleFix] deduplicate use import
[StyleFix] remove clippy warning
[BugFix] fix code related to PR-81/82 to make it works with latest changes

### CODE REVIEW

**Comment by mzanrosso** [comment]
I just saw that you pushed a new commit in the main branch, let me fix

**Comment by mzanrosso** [comment]
@jonathandturner Okay if you don't have others suggestion it's ready

File: src/vi_engine.rs [hunk 1]
```diff
@@ -43,8 +43,8 @@ impl ViEngine {
                         _ => {}
                     },
                     (Some(partial), c) => {
-                        match partial {
-                            'd' => match c {
```
Comment (sholderbach) [inline src/vi_engine.rs:47]: This lint is indicating that we need to implement more commands (Most likely through the implementation in `src/vi_parser.rs`)

File: src/painter.rs [hunk 2]
```diff
@@ -78,22 +79,13 @@ impl Painter {
...
         cursor_index_in_buffer: usize,
     ) -> Result<()> {
-        let new_index = cursor_index_in_buffer;
-        let offset = original_line.len() - new_index;
-        // Repaint logic:
-        //
-        // Start after the prompt
-        // Paint the buffer
-        // Move back to the column based on the offset
-        // Finally, reset the cursor to the saved position
```
Comment (sholderbach) [inline src/painter.rs:88]: Is this description not up to date anymore?

Comment (mzanrosso) [reply src/painter.rs:88]: I think that now it's not useful to describe the process anymore, it's auto explanatory.

File: src/vi_engine.rs (hunk 1 above)
Comment (mzanrosso) [reply src/vi_engine.rs:47]: Let me know if I have to remove this, thanks.

File: src/painter.rs (hunk 2 above)
Comment (sholderbach) [reply src/painter.rs:88]: OK

File: src/vi_engine.rs (hunk 1 above)
Comment (sholderbach) [reply src/vi_engine.rs:47]: JT started working on a more systematic parser with the last stream(s) in the other and I put my additions there. So this is not really an issue if it becomes superseded.

**Review by sholderbach — APPROVED** [review]
I don't have strong opinions on the import format. The changes look good to me.

**Comment by jonathandturner** [comment]
Looks good

**Pull request merged**